### PR TITLE
Add --namespace cli argument support for BMO

### PR DESCRIPTION
**Pull request by fmuyassarov**
Fix for #464

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -42,10 +46,12 @@ func main() {
 
 	printVersion()
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		log.Error(err, "Failed to get watch namespace")
-		os.Exit(1)
+	// override WATCH_NAMESPACE with --namespace value if provided by CLI
+	if watchNamespace == "" {
+		watchNamespace, err = k8sutil.GetWatchNamespace()
```
Comment (maelk) [inline cmd/manager/main.go]: err is undefined, you need to declare it before, or use := (that might be problematic here for the linters)

File: cmd/manager/main.go [hunk 2]
```diff
@@ -59,8 +65,8 @@ func main() {
 	opts := manager.Options{
 		LeaderElection:          true,
 		LeaderElectionID:        "baremetal-operator",
-		LeaderElectionNamespace: namespace,
-		Namespace:               namespace,
+		LeaderElectionNamespace: 	watchNamespace,
+		Namespace:               	watchNamespace,
```
Comment (maelk) [inline cmd/manager/main.go]: indentation

File: cmd/manager/main.go (hunk 1 above)
Comment (fmuyassarov) [reply cmd/manager/main.go]: yes, forgot that to add. Thanks

File: cmd/manager/main.go (hunk 2 above)
Comment (fmuyassarov) [reply cmd/manager/main.go]: resolved

File: cmd/manager/main.go (hunk 1 above)
Comment (maelk) [reply cmd/manager/main.go]: maybe at the beginning of the main function would be better, since it is used throughout the function

Comment (dhellmann) [reply cmd/manager/main.go]: Why remove the error handling here? If no namespace is provided, don't we want to exit with an error?

Comment (fmuyassarov) [reply cmd/manager/main.go]: Because if no namespace is provided then we want to watch cluster-wise (across all the namespaces).

Comment (dhellmann) [reply cmd/manager/main.go]: I thought we agreed to add the flag but still require a namespace. Why do we want the operator to ignore namespaces?

Comment (fmuyassarov) [reply cmd/manager/main.go]: If I'm not missing anything, we first check for the --namespace value in CLI and if given then override WATCH_NAMESPACE value. If --namespace not passed we check for WATCH_NAMESPACE and if not provided we watch across all namespace, no?

Comment (dhellmann) [reply cmd/manager/main.go]: You did correctly describe the behavior, as I understand it. I am surprised that we are allowing that fallback of watching all namespaces, though. I thought we said we would let the command line option override the environment variable, but exit with an error if neither were provided. Perhaps I misunderstood?

What situation do you have that makes it useful to watch all namespaces?

Comment (fmuyassarov) [reply cmd/manager/main.go]: I believe we want it to able to watch across all the namespaces in case if a user wants to create BMH in a any other namespace apart from the one where the operator itself is running. As I understood, BMO is going to be part of CAPM3, meaning that it will need to able to look for objects to reconcile in any other namespace provided by `–namespace,`  in order to fulfill the [provider contract](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html).

And, I'm not sure if we want to exit with an error if namespace is not provided in neither places but instead I believe we need to watch across all the namespace.

Comment (maelk) [reply cmd/manager/main.go]: From CAPI point of view, BMO should be able to watch all namespaces in case of a deployment that is not multi-tenant. If the deployment is for multi-tenancy, then the BMO should watch only the provided namespace. Both should be supported.

Comment (dhellmann) [reply cmd/manager/main.go]: OK, that's good information. We should capture it as part of this commit. Could you update the comment on line 50 to explain the error handling?

Comment (fmuyassarov) [reply cmd/manager/main.go]: Done.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,5 @@
 # Bare Metal Operator
 
-[![Ubuntu V1alpha2 build status](https://jenkins.nordix.org/view/Airship/job/airship_master_v1a2_integration_test_ubuntu/badge/icon?subject=Ubuntu%20E2E%20V1alpha2)](https://jenkins.nordix.org/view/Airship/job/airship_master_v1a2_integration_test_ubuntu)
-[![CentOS V1alpha2 build status](https://jenkins.nordix.org/view/Airship/job/airship_master_v1a2_integration_test_centos/badge/icon?subject=CentOS%20E2E%20V1alpha2)](https://jenkins.nordix.org/view/Airship/job/airship_master_v1a2_integration_test_centos)
```
Comment (dhellmann) [inline README.md]: Is this change related?

Comment (fmuyassarov) [reply README.md]: It is totally different thing, not related to the actual PR.  
It is just that we have dropped v1alpha2 CI now, so there will not be any CI status reported for v1alpha3 integration tests. 
So, I just wanted to fix it with the same PR as mentioned in the commit message above.

Comment (fmuyassarov) [reply README.md]: Do you want to do it on a separate PR ?

Comment (fmuyassarov) [reply README.md]: I have removed unrelated changes. Will open a separate PR for that.
